### PR TITLE
DM-6043: change mouse readout to use the MouseRadoutCntr.js and add a…

### PR DESCRIPTION
**Pull request by lznakano**
Trey,

Can you please take a look to see both the MouseReadout and the popup MouseReadout?  The popup can only be seen at the test address: http://localhost:8080/firefly/demo/ffapi-highlevel-test.html.

Thanks!

### CODE REVIEW

**Comment by xiuqin** [comment]
1. keep the names consistent between the regular Firefly readout and pop-upreadout
    World Point, Screen Point, Image Point are not on the regular readout
2. There is no way to change the coordinate system in pop-up readout
3. There is no need to display the pixel size in the pop-up readout
4. If we want the thumb nail view in the pop-up window, we should use the space to its right better.  I like the old way better. We can discuss on Monday. Please go to this URL to do a catalog search to see the readout when you put cursor over the image. http://irsa.ipac.caltech.edu/applications/Gator/

**Comment by xiuqin** [comment]
Please discard my earlier comment. I forgot to deploy the new build. Will have to check later.

**Comment by xiuqin** [comment]
1. I like the thumb nail and magnifier in the pop-up window.
2. the lock symbol is not useful since I have to click the 'x' to close the window. I think you should take it out.
3. Use the window title area for the name of the image. Currently the name does not change, always displays 'Coverage'. It should display the image name.
4.??  I want to discuss to make the window wider to have the positions next to thumb nail, the same arrangement as the regular readout. 
5. when I click on the image, the magnifier disappears. It shouldn't.  Moving cursor brings it back.

File: src/firefly/js/visualize/ui/MouseReadout.jsx [hunk 1]
```diff
@@ -470,12 +255,15 @@ function getSingleMouseReadout(plot, imagePt, toCoordinateName) {
                 break;
 
             case CoordinateSys.PIXEL:
-                var pt = plot.projection.getPixelScaleArcSec();
-                result = `  ${numeral(pt).format(precision3Digit)}"`;
+                obj =readoutItems.pixel;
+                fStr = '0.' + new Array(obj.precision + 1).join('0');
```
Comment (robyww) [inline src/firefly/js/visualize/ui/MouseReadout.jsx]: use lodash:
```
 fStr = padEnd('0.',obj.precision+2,'0')
```

File: src/firefly/js/visualize/ui/MouseReadout.jsx [hunk 2]
```diff
@@ -470,12 +255,15 @@ function getSingleMouseReadout(plot, imagePt, toCoordinateName) {
...
+                result = `${numeral(obj.value).format(fStr)} ${obj.unit || ''}`;
+
                 break;
             case CoordinateSys.SCREEN_PIXEL:
-                var size = plot.projection.getPixelScaleArcSec() / plot.zoomFactor;
-                result = `  ${numeral(size).format(precision3Digit)}"`;
+                 obj =readoutItems.screenPixel;
+                fStr = '0.' + new Array(obj.precision + 1).join('0');
```
Comment (robyww) [inline src/firefly/js/visualize/ui/MouseReadout.jsx]: like line 259, use lodash

File: src/firefly/js/visualize/ui/MouseReadout.jsx [hunk 3]
```diff
@@ -83,364 +77,155 @@ const column5 = {
...
-    else {
-        coordinate = coordinateMap[coordinateRadioValue];
-        //if coordinate is not define, assign it as below
-        if (!coordinate) coordinate = CoordinateSys.UNDEFINED;
+    var fluxValueArrays=[];
+    var fluxLabelArrays=[];
+    var fluxValue,  formatStr;
+
+    for (let i = 0; i < fluxObj.length; i++) {
+            formatStr = '0.' + new Array(fluxObj[i].precision + 1).join('0');
```
Comment (robyww) [inline src/firefly/js/visualize/ui/MouseReadout.jsx]: like line 259, use lodash

File: src/firefly/js/visualize/ui/MouseReadout.jsx [hunk 4]
```diff
@@ -470,12 +255,15 @@ function getSingleMouseReadout(plot, imagePt, toCoordinateName) {
                 break;
 
             case CoordinateSys.PIXEL:
-                var pt = plot.projection.getPixelScaleArcSec();
-                result = `  ${numeral(pt).format(precision3Digit)}"`;
+                obj =readoutItems.pixel;
+                fStr = '0.' + new Array(obj.precision + 1).join('0');
+                result = `${numeral(obj.value).format(fStr)} ${obj.unit || ''}`;
+
```
Comment (robyww) [inline src/firefly/js/visualize/ui/MouseReadout.jsx]: Picky: you could break line 259 and 260 into a simple function as reuse it several times.
_something like this:_
```
const myFormat= (v,precision) => numeral(v).format(padEnd('0.',precision+2,'0') );
```

**Comment by robyww** [comment]
API popup readout:

- The title should be the title of the plot not 'coverage'
- We do need to add an option to disable the magnifier and thumbnail since gator does not use them. We can talk about that
- Remember there needs to be a full readout and a small one for the popup.  I don't see that you have yet done the full one.
-

**Comment by robyww** [comment]
It is looking good.  The mouse readout is a long cleaner. there is still a little more work to be done. We can talk about that on Monday.

**Comment by lznakano** [comment]
Xiuqin:


2.  I think that I need to disable the x in the window.  The lock is to 
make the window not to be hidden and not to be closed.  I will fix this.

3.  The gator mouse readout popup has the "Coverage"  for every plot.  I 
asked Angela and she referred this readouts as a Coverage. I always use 
the GWT version as a requirement to do the work.
     The regular mouse readout did use title.   I changed it to 
"Coverage" on purpose.  I will fix this.

In general, I think  it  will be more efficient and useful if any 
features to be changed and to be improved in the conversion are added to 
the JIRA  ticket.   I always refer both the JIRA ticket and the GWT.



On 6/3/16 10:22 PM, Xiuqin Wu wrote:
>
>  1. I like the thumb nail and magnifier in the pop-up window.
>  2. the lock symbol is not useful since I have to click the 'x' to
>     close the window. I think you should take it out.
>  3. Use the window title area for the name of the image. Currently the
>     name does not change, always displays 'Coverage'. It should
>     display the image name. 4.?? I want to discuss to make the window
>     wider to have the positions next to thumb nail, the same
>     arrangement as the regular readout.
>  4. when I click on the image, the magnifier disappears. It shouldn't.
>     Moving cursor brings it back.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/97#issuecomment-223737398>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ALbTfWEt1n4TO7hGV2_wgiTqIB5D2Tynks5qIQuhgaJpZM4It9xa>.
>

-- 
Lijun Zhang

LSST/IPAC
California Institute of Technology
770 S. Wilson Ave, Mail Code 100-22
Pasadena, CA 91125, USA
Voice: 1-626-395-1972 Fax:1-626-395-1816

File: src/firefly/js/visualize/ui/MouseReadout.jsx (hunk 4 above)
Comment (lznakano) [inline src/firefly/js/visualize/ui/MouseReadout.jsx]: Great suggestion!!! I like it.


On 6/4/16 7:22 AM, Trey Roby wrote:
> |const myFormat= (v,precision) => 
> numeral(v).format(padEnd('0.',precision+2,'0') );|

-- 
Lijun Zhang

LSST/IPAC
California Institute of Technology
770 S. Wilson Ave, Mail Code 100-22
Pasadena, CA 91125, USA
Voice: 1-626-395-1972 Fax:1-626-395-1816

**Comment by lznakano** [comment]
Four types of readouts are checked in and they can be tested using the html file below.   

http://localhost:8080/firefly/demo/mouseReadoutMinimalTest1.html
http://localhost:8080/firefly/demo/mouseReadoutMinimalTest2.html
http://localhost:8080/firefly/demo/mouseReadoutFullTest1.html
http://localhost:8080/firefly/demo/mouseReadoutFullTest2.html

Trey, 
Can you please help me to check why the thumbnail's width shirked to 61pt instead of 70 pt?  I can not figure which event controls it. 

Thanks!

**Comment by xiuqin** [comment]
It is almost there. 
On my monitor, the space between position display and lock needs a few more pixel space. 
The space between thumbnail and magnifier look different in the two readouts below: http://localhost:8080/firefly/demo/mouseReadoutMinimalTest2.html and http://localhost:8080/firefly/demo/mouseReadoutFullTest2.html

**Pull request merged**